From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 22 Apr 2020 20:34:02 +0200
Subject: core: make sure to restore the control command id, too

Fixes: #15356
(cherry picked from commit e9da62b18af647bfa73807e1c7fc3bfa4bb4b2ac)
---
 src/core/service.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/core/service.c b/src/core/service.c
index 5f5bcb3..eb66884 100644
--- a/src/core/service.c
+++ b/src/core/service.c
@@ -2689,9 +2689,10 @@ static int service_deserialize_exec_command(Unit *u, const char *key, const char
                                 break;
         }
 
-        if (command && control)
+        if (command && control) {
                 s->control_command = command;
-        else if (command)
+                s->control_command_id = id;
+        } else if (command)
                 s->main_command = command;
         else
                 log_unit_warning(u, "Current command vanished from the unit file, execution of the command list won't be resumed.");
